Sources/Widgets/Widgets3D/SplineWidget/behavior.js

```diff
@@ -317,24 +312,24 @@ export default function widgetBehavior(publicAPI, model) {
 
   publicAPI.handleKeyDown = ({ key }) => {
     model.keysDown[key] = true;
-    publicAPI.updateResolution();
+
+    if (!model.hasFocus) {
+      return;
+    }
 
     if (key === 'Enter') {
       if (model.widgetState.getHandleList().length > 0) {
-        publicAPI.invokeEndInteractionEvent();
-        model.interactor.render();
-      }
-    } else if (model.hasFocus) {
-      if (key === 'Escape') {
-        publicAPI.reset();
         publicAPI.loseFocus();
-      } else if (key === 'Delete' || key === 'Backspace') {
-        if (model.lastHandle) {
-          model.widgetState.removeHandle(model.lastHandle);
+      }
+    } else if (key === 'Escape') {
+      publicAPI.reset();
+      publicAPI.loseFocus();
```

Comment:
@finetjul @floryst what do you think about using escape to only complete (loseFocus, which calls `invokeEndInteractionEvent`) without resetting/hiding the representation, and leave it to the application to reset it if it wants by listening to that callback.

Then we would have it lose focus on `Escape`, and maybe `Enter` as well (though `Enter` isn't used in other widgets AFAIK)? If we wanted to configure something on this widget, I'd suggest that would be a list of keys that would end the interaction (defaulting to `['Escape']`).

Reply:
@floryst : I talked with @finetjul and he'd like to keep the current behavior as a default, at least for now, until we maybe add some kinf of "Action Manager" to configure behavior (and not just representation). I'm fine with that, and keeping this as is.

Reply:
I'm also okay with this then for now. It would be nice to have this customizable from an application perspective in the future.